Migrate Separator and Spacer to SCSS modules

vineethasok opened the pull request:
Refactored Separator and Spacer components to use SCSS modules for styling instead of styled-components. Added new SCSS files, updated Storybook configuration and preview to support SCSS, and introduced a script to generate a filtered CSS theme for SCSS-migrated components. Updated package.json, .gitignore, and index.ts to ensure proper CSS variable imports and side effects. Also added utility functions and mixins for SCSS, and improved Storybook controls for these components.

Note: This is the first step to migrating components to scss one

vineethasok committed:
Migrate Separator and Spacer to SCSS modules

Refactored Separator and Spacer components to use SCSS modules for styling instead of styled-components. Added new SCSS files, updated Storybook configuration and preview to support SCSS, and introduced a script to generate a filtered CSS theme for SCSS-migrated components. Updated package.json, .gitignore, and index.ts to ensure proper CSS variable imports and side effects. Also added utility functions and mixins for SCSS, and improved Storybook controls for these components.

vineethasok committed:
Improve Storybook theme support and SCSS theme generation

Refactored ThemeBlock in Storybook to accept a theme prop and set color-scheme for better light/dark support. Updated SCSS theme generation script to always include Storybook-specific variables. Added --click-storybook-global-background to SCSS theme and improved preview styles for better layout handling.

vineethasok committed:
Set color-scheme and data-theme on document root

Added a useEffect to update the document root's color-scheme and data-theme attributes based on the selected theme. This enables support for the CSS light-dark() function and improves theme handling.

vineethasok committed:
Fix the chromatic UI diff error

gjones reviewed (approved):
Overall functionality is good. So 👍 

There's quite a few changes to the build-tokens.js file in here, I'm unsure why they would be in with the separator and spacer updates. What do they do?

vineethasok committed:
Merge branch 'main' into scss-infra

vineethasok commented on build-tokens.js:
```diff
@@ -51,7 +77,7 @@ StyleDictionary.registerFormat({
 
 StyleDictionary.extend({
-  source: [`./tokens/**/!(${themes.join("|*.")}).json`],
+  source: [`./tokens/**/*.json`],
```
We could actually remove the css from being build as we are not using it at all 
Also separate the logic for variable json and the types to 2 calls which would not cause any problems at all

vineethasok committed:
Merge branch 'main' into scss-infra

vineethasok committed:
Update dependencies in yarn.lock

Regenerated yarn.lock to update and add new package versions, reflecting changes in project dependencies. This may include upgrades, security patches, or compatibility improvements.

vineethasok committed:
Refactor Separator and Spacer to use tokens variables

Updated SCSS files for Separator and Spacer components to replace CSS custom properties with SCSS tokens variables for consistency and maintainability. Added data-cui attributes to both components for improved DOM identification and testing.

vineethasok committed:
Revert unnecessary changes

vineethasok committed:
Revert unnecessary changes

vineethasok committed:
Migrate global and app styles from CSS to SCSS

Removed legacy CSS files and replaced them with SCSS modules for global and app styles. Updated imports and file references throughout the codebase to use SCSS. Deleted the generate-scss-theme script and filtered theme CSS, as SCSS modules now handle component styling. Updated documentation to clarify CSS import requirements and migration status.

vineethasok committed:
Migrate global CSS to SCSS modules and update config

Removed legacy global CSS files and updated styles to use SCSS modules. Updated Storybook and Vite configuration for modern SCSS compiler support. Cleaned up README and code references to legacy CSS imports, and improved VSCode TypeScript settings.

vineethasok committed:
Merge branch 'main' into scss-infra

vineethasok committed:
Revert unwanted changes

vineethasok committed:
Remove redundant SCSS config from Storybook setup

Moved SCSS preprocessor configuration to root vite.config.ts to prevent CSS injection order issues with styled-components. Storybook now inherits SCSS settings from the main Vite config.

vineethasok committed:
Fix CSS specificity issues in Storybook config

Set `cssCodeSplit` to false in Storybook Vite config to ensure styled-components styles are injected after CSS modules. This resolves specificity problems when using the polymorphic 'as' prop in components.

vineethasok commented on .storybook/main.ts:
```diff
@@ -68,8 +68,11 @@ const config: StorybookConfig = {
+    // Ensure styled-components styles are injected after CSS modules
+    // This fixes CSS specificity issues when components use polymorphic 'as' prop
+    config.build = config.build || {};
+    config.build.cssCodeSplit = false;
```
This is only for the storybook and not for our vite build

ariser reviewed (approved):
lgtm in general, but I'm not sure why we need these variant mixins
